src/storage/storage.cpp

```diff
@@ -308,7 +308,7 @@ void Storage::PopulateLayout(DataLayout &layout)
     }
 
     {
-        io::FileReader hsgr_file(config.hsgr_data_path, io::FileReader::HasNoFingerprint);
+        io::FileReader hsgr_file(config.hsgr_data_path, io::FileReader::VerifyFingerprint);
```

Comment:
Yeah, it does.  If you take a look at the changes I made to `include/storage/serialization.hpp`, you'll see that the fingerprint reading logic was duplicated there.  I removed that copied logic, so now the `.hsgr` loading can use the same fingerprint validation logic that everyone else does.

Slowly but surely removing all the mess.....